api_specifications/evidence-api-1.0.0.yaml

```diff
@@ -0,0 +1,186 @@
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/EvidenceRequest'
+        required: true
+      responses:
+        '201':
+          description: ReponseInfo with case(s) updated sudcessfully
+          content:
+            '*/*':
+              schema:
+                $ref: '#/components/schemas/EvidenceResponse'
+        '400':
+          description: Case(s) creation failed
+          content:
+            '*/*':
+              schema:
+                $ref: >-
+                  https://raw.githubusercontent.com/egovernments/DIGIT-OSS/master/core-services/docs/common-contract_v1-1.yml#/components/schemas/ErrorRes
+      x-codegen-request-body-name: caseRequest
+  /artifacts/v1/_update:
+    post:
+      tags:
+        - Evidence
+      summary: update case requests.
+      requestBody:
+        description: Details for the new case(s) + RequestInfo meta data.
+        content:
+          '*/*':
+            schema:
+              $ref: >-
+                https://raw.githubusercontent.com/egovernments/DIGIT-OSS/master/core-services/docs/common-contract_v1-1.yml#/components/schemas/EvidenceRequest
+        required: true
+      responses:
+        '201':
+          description: ReponseInfo with case(s) updated successfully
+          content:
+            '*/*':
+              schema:
+                $ref: >-
+                  https://raw.githubusercontent.com/egovernments/DIGIT-OSS/master/core-services/docs/common-contract_v1-1.yml#/components/schemas/EvidenceResponse
+        '400':
+          description: Case(s) updation failed
+          content:
+            '*/*':
+              schema:
+                $ref: >-
+                  https://raw.githubusercontent.com/egovernments/DIGIT-OSS/master/core-services/docs/common-contract_v1-1.yml#/components/schemas/ErrorRes
+
+components:
+  schemas:
+    Artifact:
+      required: [tenantId]
+      oneOf:
+        - required: [filingNumber, cnrNumber]
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+          description: This id is autogenerated id of the artifact. It is the primary key
+          readOnly: true
+          minLength: 2
+          maxLength: 64
+        tenantId:
+          type: string
+          description: This is tenantId of the case based on the state.
+        artifactNumber:
+          type: string
+          description: Identification number assigned when an artifact is uploaded
+          minLength: 2
+          maxLength: 64
+        evidenceNumber:
+          type: string
+          description: Formatted evidence number assigned after evidence is accepted by the deciding authority.
+          minLength: 2
+          maxLength: 64
+        externalRefNumber:
+          type: string
+          description: Reference to a physical artifact or other evidence collected during the course of an investigation
+          minLength: 2
+          maxLength: 128
```

Comment:
external reference may be longer string. we don't know. suggest we make this 256